api/src/main/java/io/neba/api/resourcemodels/AnnotatedFieldMapper.java

```diff
@@ -0,0 +1,162 @@
+ * </p>
+ * <p>
+ * For example, the following service would be responsible for fields annotated with &#64;MyAnnotation
+ * with a type <em>assignable to</em> {@link java.util.Collection}, e.g. List&lt;Resource&gt; or Set&lt;String&gt;.
+ * </p>
+ * <pre>
+ * &#64;Service
+ * &#64;Component(immediate = true)
+ * public class MyFieldMapper&lt;Collection, MyAnnotation&gt; {
+ *     public Class&lt;? extends Collection&gt; getFieldType() {
+ *         return Collection.class;
+ *     }
+ *
+ *     public Class&lt;Annotation&gt; getAnnotationType() {
+ *         return MyAnnotation.class;
+ *     }
+ *
+ *     public Collection map(OngoingMapping&lt;Collection, MyAnnotation&gt; ongoingMapping) {
+ *         ...
+ *     }
+ * }
+ * </pre>
+ * <p>
+ * Custom mappers are always invoked <em>after</em> all of NEBA's standard mappings have occurred, but before
+ * the corresponding value was set on the model's field. They may thus make use of the already resolved
+ * value or choose to provide a different one.
+ * </p>
+ *
+ * <p>
+ *     <strong>It is crucial for a {@link AnnotatedFieldMapper} to always return a value that is assignment compatible
+ *     to the {@link OngoingMapping#getFieldType() field type}, i.e. either of the same or of a more specific type.</strong>
+ *     It is insufficient to return a type compatible to the {@link #map(AnnotatedFieldMapper.OngoingMapping) mapping methods}
+ *     return type declaration. This return type only represents the type any returned value must be compatible to.<br />
+ *     For instance, if a mapper is responsible for {@link java.util.Collection}, it must take care to return the field's actual
+ *     collection type, e.g. {@link java.util.List} or {@link java.util.Set}. Otherwise, an exception will arise.
+ * </p>
+ *
+ * <p>
+ *     Implementations <strong>must never</strong> store any contextual data provided by the
+ *     {@link io.neba.api.resourcemodels.AnnotatedFieldMapper.OngoingMapping}
+ *     as this data stems from arbitrary OSGi bundles with independent life cycles.
+ *     Storing any data would result in a class loader / memory leak when these bundles change.
+ * </p>
+ *
+ * @param <FieldType> the super type of the {@link java.lang.reflect.Field#getType() field type}
+ *        of the mapped type.
+ * @param <AnnotationType> the exact type of the annotation this mapper is responsible for.
+
+ * @author Olaf Otto
+ */
+public interface AnnotatedFieldMapper<FieldType, AnnotationType extends Annotation> {
+    /**
+     * Represents the contextual data of a field mapping during a resource to model mapping.
+     *
+     * @author Olaf Otto
+     */
+    public interface OngoingMapping<FieldType, AnnotationType> {
+        /**
+         * @return The currently resolved value of the field,
+         * or <code>null</code> if no value could be resolved for the field.
+         */
+        FieldType getResolvedValue();
```

Comment:
So this is a collection of the correct type in all cases, if FieldType is a collection?